### ISSUE TITLE
Wrong linking / validation when auto build disabled.

### ISSUE THREAD

**Issue by baemapo** [eclipse/xtext#1151]
Having auto build disabled if you manually change i.E. a variable name then the validation will place markers on files referencing the old variable's name. If you save the change the DirtaStateAwareDescriptions will get discarded as the build is supposed to be triggered. As that is not the case the referencing editors get when validating the previously saved state (from the last build) and thus see the old variable there removing all markers.

I would like to contribute a change that I implemented, which makes the DirtyStateAwareDescriptions not to be discarded unless an event is received indicating that the index has been built for the affected ressource.

**Comment by cdietrich** [comment]
sounds like https://bugs.eclipse.org/bugs/show_bug.cgi?id=491214

**Comment by baemapo** [comment]
Yes, it seems to basically be the same behavior. In my case I don't get any exception, my error markers are simply gone, as the linker thinks the old variable is still available.
